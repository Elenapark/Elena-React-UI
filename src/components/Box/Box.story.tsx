import React, { FC } from 'react';
import { Meta } from '@storybook/react';
import { Box, BoxProps } from './Box';

export default {
  title: 'Box',
  component: Box,
} as Meta<BoxProps>;

export const Default: FC = () => {
  return <Box>hello world</Box>;
};
