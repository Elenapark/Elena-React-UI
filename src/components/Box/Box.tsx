import styled from 'styled-components';
import { space, layout, color, flexbox, SpaceProps, LayoutProps, ColorProps, FlexboxProps } from 'styled-system';
import React from 'react';

const StyledBox = styled.div`
  ${space}
  ${layout}
  ${color}
  ${flexbox}
`;

export type BoxProps = SpaceProps & LayoutProps & ColorProps & FlexboxProps;

const Box: FCC<BoxProps> = (props: any) => {
  return <StyledBox {...props} data-testid="box-comp" />;
};

export default Box;
