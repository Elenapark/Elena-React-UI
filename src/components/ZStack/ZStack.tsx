import styled from 'styled-components';
import cx from 'classnames';
import React, { ReactNode } from 'react';
import { color, ColorProps, flexbox, FlexboxProps, layout, LayoutProps, space, SpaceProps } from 'styled-system';
import useRefinedChildren from '../../hooks/use-refined-children';

interface CustomZStackProps {
  className?: string;
  children?: ReactNode;

  [x: string]: any;
}

export type ZStackProps = CustomZStackProps & SpaceProps & LayoutProps & ColorProps & FlexboxProps;

export const ZStack: FCC<ZStackProps> = ({ className, children, ...props }) => {
  const refinedChildren = useRefinedChildren({
    children,
    className,
  });

  return (
    <StyledZStack data-testid="zstack-comp" className={cx('zstack', className)} {...props}>
      {refinedChildren}
    </StyledZStack>
  );
};

const StyledZStack = styled.div<ZStackProps>`
  ${space}
  ${layout}
  ${color}
  ${flexbox}
  position: relative;
`;
