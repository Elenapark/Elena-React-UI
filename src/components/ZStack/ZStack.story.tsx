import React, { FC } from 'react';
import { Meta } from '@storybook/react';
import ZStack, { CustomZStackProps } from './ZStack';
import Box from '../Box/Box';

const COLOR = ['red', 'orange', 'yellow', 'green', 'blue', 'purple'];

export default {
  title: 'ZStack',
  component: ZStack,
} as Meta<CustomZStackProps>;

export const Default: FC = () => {
  return (
    <ZStack>
      {COLOR.map((_, idx) => {
        return (
          <Box bg={COLOR[idx]} position="absolute" top={idx * 10} left={idx * 10} width={100} height={100}>
            {`box${idx}`}
          </Box>
        );
      })}
    </ZStack>
  );
};
