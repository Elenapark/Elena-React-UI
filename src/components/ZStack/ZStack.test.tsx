import { render, screen } from '@testing-library/react';
import {Box} from '../Box/Box';
import React from 'react';
import {ZStack} from './ZStack';

const COLOR = ['red', 'orange', 'yellow', 'green', 'blue', 'purple'];

const children = COLOR.map((_, idx) => {
  return (
    <Box
      bg={COLOR[idx]}
      position="absolute"
      top={idx * 10}
      left={idx * 10}
      width={100}
      height={100}
      data-testid="zstack-child-comp"
    >
      {`box${idx}`}
    </Box>
  );
});

const renderStack = () => {
  const defaultProps = {
    className: 'test',
    children,
  };
  render(<ZStack {...defaultProps} />);
};

describe('<ZStack  />', () => {
  it('should render', () => {
    renderStack();
    expect(screen.getByTestId('zstack-comp')).toBeInTheDocument();
  });

  it('should render children', () => {
    renderStack();
    expect(screen.getAllByTestId('zstack-child-comp')).toHaveLength(6);
  });

  it('should have proper attribute', () => {
    renderStack();
    COLOR.map((_, idx) => {
      return expect(screen.getByText(`box${idx}`)).toHaveAttribute('style', `z-index: ${idx};`);
    });
  });
});
