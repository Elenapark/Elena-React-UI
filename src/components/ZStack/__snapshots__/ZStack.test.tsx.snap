// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ZStack  /> should render 1`] = `
<div
  class="sc-gswNZR grmAwb zstack test"
  data-testid="zstack-comp"
>
  <div
    class="sc-bcXHqe ga-dMOh"
    data-testid="zstack-child-comp"
    height="100"
    style="z-index: 0;"
    width="100"
  >
    box0
  </div>
  <div
    class="sc-bcXHqe VNdXa"
    data-testid="zstack-child-comp"
    height="100"
    style="z-index: 1;"
    width="100"
  >
    box1
  </div>
  <div
    class="sc-bcXHqe fjWdZK"
    data-testid="zstack-child-comp"
    height="100"
    style="z-index: 2;"
    width="100"
  >
    box2
  </div>
  <div
    class="sc-bcXHqe BTrBJ"
    data-testid="zstack-child-comp"
    height="100"
    style="z-index: 3;"
    width="100"
  >
    box3
  </div>
  <div
    class="sc-bcXHqe btohpo"
    data-testid="zstack-child-comp"
    height="100"
    style="z-index: 4;"
    width="100"
  >
    box4
  </div>
  <div
    class="sc-bcXHqe ducREs"
    data-testid="zstack-child-comp"
    height="100"
    style="z-index: 5;"
    width="100"
  >
    box5
  </div>
</div>
`;
